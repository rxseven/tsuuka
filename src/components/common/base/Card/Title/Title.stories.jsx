import React from 'react';
import { storiesOf } from '@storybook/react';

import Title from './index';

const title = 'Components/Common/Base/Card/Title';
const component = <Title>Card Title</Title>;

storiesOf(title, module).add('default', () => component);
