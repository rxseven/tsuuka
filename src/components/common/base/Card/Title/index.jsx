import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';

const propTypes = exact({
  children: PropTypes.node.isRequired
});

function Title({ children }) {
  return <h5 className="card-title">{children}</h5>;
}

Title.propTypes = propTypes;

export default Title;
