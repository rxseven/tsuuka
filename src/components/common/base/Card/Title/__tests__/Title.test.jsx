import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Title from '../index';

describe('<Card.Title />', () => {
  // Arrange
  const props = { children: 'content' };
  const component = <Title {...props} />;

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
