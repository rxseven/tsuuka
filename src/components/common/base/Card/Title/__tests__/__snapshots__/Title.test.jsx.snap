// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card.Title /> Snapshot tests should render correctly 1`] = `
<h5
  className="card-title"
>
  content
</h5>
`;
